specs/MultiProfile.md

```diff
@@ -0,0 +1,289 @@
+
+## Provide options to create WebView2 with a specific profile
+
+### Win32 C++
+
+```cpp
+HRESULT AppWindow::CreateControllerWithOptions()
+{
+    auto webViewEnvironment4 =
+        m_webViewEnvironment.try_query<ICoreWebView2Environment4>();
+    if (!webViewEnvironment4)
+    {
+        FeatureNotAvailable();
+        return S_OK;
+    }
+
+    Microsoft::WRL::ComPtr<ICoreWebView2ControllerOptions> options;
+    HRESULT hr = webViewEnvironment4->CreateCoreWebView2ControllerOptions(
+        m_webviewOption.profile.c_str(), m_webviewOption.isInPrivate, options.GetAddressOf());
+    if (hr == E_INVALIDARG)
+    {
+        ShowFailure(hr, L"Unable to create WebView2 due to an invalid profile name.");
+        CloseAppWindow();
+        return S_OK;
+    }
+    CHECK_FAILURE(hr);
+
+#ifdef USE_WEBVIEW2_WIN10
+    if (m_dcompDevice || m_wincompCompositor)
+#else
+    if (m_dcompDevice)
+#endif
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2CompositionControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2CompositionControllerCompletedHandler>(
+                [this](
+                    HRESULT result,
+                    ICoreWebView2CompositionController* compositionController) -> HRESULT {
+                    auto controller =
+                        wil::com_ptr<ICoreWebView2CompositionController>(compositionController)
+                            .query<ICoreWebView2Controller>();
+                    return OnCreateCoreWebView2ControllerCompleted(result, controller.get());
+                })
+                .Get()));
+    }
+    else
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2ControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2ControllerCompletedHandler>(
+                this, &AppWindow::OnCreateCoreWebView2ControllerCompleted)
+                .Get()));
+    }
+
+    return S_OK;
+}
+```
+
+## Access the profile property of WebView2
+
+### Win32 C++
+
+```cpp
+HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICoreWebView2Controller* controller)
+{
+    // ...
+
+    m_controller = controller;
+    
+    // Gets the webview object from controller.
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+
+    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
+    if (webview7)
+    {
+        // Gets the profile property of webview.
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webview7->get_Profile(&profile));
+
+        // Accesses the profile object.
+        wil::unique_cotaskmem_string name;
+        CHECK_FAILURE(profile->get_ProfileName(&name));
+        BOOL inPrivateModeEnabled = FALSE;
+        CHECK_FAILURE(profile->get_IsInPrivateModeEnabled(&inPrivateModeEnabled));
+        wil::unique_cotaskmem_string profile_path;
```

Comment:
No need to get profile path which isn't used.

Reply:
Actually we're displaying the profile path (only the last folder name, not full path) but not the profile name :)

The |m_profileName| is a little bit confusing, renamed it to |m_profileDirName|. Then removed  `profile->get_ProfileName()` which is not used here.